packages/service-worker/worker/src/driver.ts

```diff
@@ -490,6 +490,16 @@ export class Driver implements Debuggable, UpdateSource {
         table.read<LatestEntry>('latest'),
       ]);
 
+      // Make sure latest manifest is correctly installed. If not (e.g. corrupted data),
+      // it could stay locked in EXISTING_CLIENTS_ONLY or SAFE_MODE state.
+      if (!this.versions.has(latest.latest) &&
+          Object.keys(manifests).indexOf(latest.latest) === -1) {
```

Comment:
Looking at this again, why not:

```suggestion
          !manifests.hasOwnProperty(latest.latest)) {
```

:grin:

Reply:
Even better :+1: 

PR updated!